src/components/my-widgets-settings-dialog.jsx

```diff
@@ -0,0 +1,728 @@
+import React, { useState, useEffect } from 'react'
+import Modal from './modal'
+import fetchOptions from '../util/fetch-options'
+import DatePicker from "react-datepicker"
+import "./my-widgets-settings-dialog.scss"
+import "react-datepicker/dist/react-datepicker.css"
+
+const fetchUpdateWidget = (args) => fetch('/api/json/widget_instance_update', fetchOptions({body: `data=${encodeURIComponent(JSON.stringify(args))}`}))
+const fetchUsers = (arrayOfUserIds) => fetch('/api/json/user_get', fetchOptions({body: `data=${encodeURIComponent(JSON.stringify([arrayOfUserIds]))}`}))
+
+const initForm = {
```

Comment:
be cautious about using default/initial settings like this.
since this is pass by ref, if the component changes properties of these they may bleed to new defaults 

One defense is to create a function that returns a brand new object instead

```javascript
const defaultForm = () => ({
  key: value
})

...

const [formData, setFormData] = useState(defaultForm())


```